lib/nimble_options.ex

```diff
@@ -590,104 +601,127 @@ defmodule NimbleOptions do
        when not (value == :infinity or (is_integer(value) and value >= 0)) do
     error_tuple(
       key,
       value,
-      "invalid value for #{render_key(key)}: expected non-negative integer or :infinity, got: #{inspect(value)}"
+      "invalid value for #{render_key(key)}: expected non-negative integer or :infinity",
+      ", got: #{inspect(value)}",
+      redact
     )
   end
 
-  defp validate_type(:string, key, value) when not is_binary(value) do
+  defp validate_type(:string, key, value, redact) when not is_binary(value) do
     error_tuple(
       key,
       value,
-      "invalid value for #{render_key(key)}: expected string, got: #{inspect(value)}"
+      "invalid value for #{render_key(key)}: expected string",
+      ", got: #{inspect(value)}",
+      redact
     )
   end
 
-  defp validate_type(:boolean, key, value) when not is_boolean(value) do
+  defp validate_type(:boolean, key, value, redact) when not is_boolean(value) do
     error_tuple(
       key,
       value,
-      "invalid value for #{render_key(key)}: expected boolean, got: #{inspect(value)}"
+      "invalid value for #{render_key(key)}: expected boolean",
+      ", got: #{inspect(value)}",
+      redact
     )
   end
 
-  defp validate_type(:keyword_list, key, value) do
+  defp validate_type(:keyword_list, key, value, redact) do
     if keyword_list?(value) do
       {:ok, value}
     else
       error_tuple(
         key,
         value,
-        "invalid value for #{render_key(key)}: expected keyword list, got: #{inspect(value)}"
+        "invalid value for #{render_key(key)}: expected keyword list",
+        ", got: #{inspect(value)}",
+        redact
       )
     end
   end
 
-  defp validate_type(:non_empty_keyword_list, key, value) do
+  defp validate_type(:non_empty_keyword_list, key, value, redact) do
     if keyword_list?(value) and value != [] do
       {:ok, value}
     else
       error_tuple(
         key,
         value,
-        "invalid value for #{render_key(key)}: expected non-empty keyword list, got: #{inspect(value)}"
+        "invalid value for #{render_key(key)}: expected non-empty keyword list",
+        ", got: #{inspect(value)}",
+        redact
       )
     end
   end
 
-  defp validate_type(:map, key, value) do
-    validate_type({:map, :atom, :any}, key, value)
+  defp validate_type(:map, key, value, redact) do
+    validate_type({:map, :atom, :any}, key, value, redact)
   end
 
-  defp validate_type({:map, key_type, value_type}, key, map) when is_map(map) do
+  defp validate_type({:map, key_type, value_type}, key, map, redact) when is_map(map) do
     map
     |> Enum.reduce_while([], fn {key, value}, acc ->
-      with {:ok, updated_key} <- validate_type(key_type, {__MODULE__, :key}, key),
-           {:ok, updated_value} <- validate_type(value_type, {__MODULE__, :value, key}, value) do
+      with {:ok, updated_key} <-
+             validate_type(key_type, {__MODULE__, :key}, key, false),
```

Comment:
@whatyouhide I'm not redacting keys. I think this is the correct approach but just wanted to point it out for review.

Reply:
Yeah we should not redact keys.

Reply:
That's what I though too. Thanks for confirming!